contracts/StMATIC.sol

```diff
@@ -898,4 +897,59 @@ contract StMATIC is
 
         return (withdrawExchangeRate * unbond.shares) / exchangeRatePrecision;
     }
+
+    /**
+     * @dev Returns the number of operators from operatorShares that have the delegate flag enabled
+     */
+    function getOperatorsWithDelegationEnabledCount(
+        Operator.OperatorInfo[] memory operatorShares
+    ) private view returns (uint256) {
+        uint256 count;
+        uint256 operatorSharesLength = operatorShares.length;
+
+        for (uint256 i = 0; i < operatorSharesLength; i++) {
+            IValidatorShare validatorShare = IValidatorShare(
+                operatorShares[i].validatorShare
+            );
+            if (validatorShare.delegation()) {
+                count++;
+            }
+        }
+
+        return count;
+    }
+
+    /**
+     * @dev Get OperatorInfos from operators that have delegation flag set to true
+     */
+    function getOperatorsWithDelegationEnabled()
+        private
+        view
+        returns (Operator.OperatorInfo[] memory)
+    {
+        Operator.OperatorInfo[] memory operatorShares = nodeOperatorRegistry
+            .getOperatorInfos(true);
+        uint256 operatorSharesLength = operatorShares.length;
+
+        uint256 feasibleOperatorsCount = getOperatorsWithDelegationEnabledCount(
```

Comment:
You solution works fine, but I have another one that allows us to make one cross contract call per operator.

you don't need to calculate the `count`
1- You can create an array with same length `operatorShares`
2- Loop through and check if the delegation flag set to true.
3- compare if feasibleOperatorsIndex is equal operatorShares if not
pop the empty elements from the table.

I had the same issue in the getOperatorInfo see here how I fixed it
https://github.com/Shard-Labs/PoLido/blob/bda8645bff4916a12d21fff38df8379e036724fd/contracts/NodeOperatorRegistry.sol#L980

Reply:
So what you did there is:
1. Create a new array of structs of size `n`
2. Loop through the old array, take the feasible elements and add them to the new array
3. Create a new array that is the right size and copy the elements to it
There are 2 for loops in total.

I use 1 for loop to calculate the right size for array and 1 for loop to add the feasible elements to array.
I am not sure if this will be an improvement.